x/twap/store.go

```diff
@@ -162,7 +161,11 @@ func (k Keeper) getRecordAtOrBeforeTime(ctx sdk.Context, poolId uint64, t time.T
 	}
 
 	for _, twap := range twaps {
-		if twap.Asset0Denom == asset0Denom && twap.Asset1Denom == asset1Denom {
+		if twap.Asset0Denom != asset0Denom {
+			return types.TwapRecord{}, fmt.Errorf(" Provided asset0denom (%s) does not match twap.Asset0Denom (%s)", asset0Denom, twap.Asset0Denom)
+		} else if twap.Asset1Denom != asset1Denom {
+			return types.TwapRecord{}, fmt.Errorf(" Provided asset1denom (%s) does not match twap.Asset1Denom (%s)", asset1Denom, twap.Asset1Denom)
+		} else {
```

Comment:
While this achieves the same result as what it replaces, it:
1) now passes mutation tests
2) outputs specific errors instead of grouping it into one